isisd/isis_dynhn.c

```diff
@@ -42,30 +42,29 @@
 
 DEFINE_MTYPE_STATIC(ISISD, ISIS_DYNHN, "ISIS dyn hostname");
 
-extern struct host host;
-
-struct list *dyn_cache = NULL;
 static int dyn_cache_cleanup(struct thread *);
 
 void dyn_cache_init(struct isis *isis)
 {
-	if (dyn_cache == NULL)
-		dyn_cache = list_new();
+	isis->dyn_cache = list_new();
 	if (!CHECK_FLAG(im->options, F_ISIS_UNIT_TEST))
 		thread_add_timer(master, dyn_cache_cleanup, isis, 120,
 				 &isis->t_dync_clean);
-	return;
 }
 
-void dyn_cache_cleanup_all(void)
+void dyn_cache_finish(struct isis *isis)
 {
 	struct listnode *node, *nnode;
 	struct isis_dynhn *dyn;
 
-	for (ALL_LIST_ELEMENTS(dyn_cache, node, nnode, dyn)) {
-		list_delete_node(dyn_cache, node);
+	thread_cancel(&isis->t_dync_clean);
```

Comment:
Please add a null check as this timer event isn't scheduled when `F_ISIS_UNIT_TEST` is set (or just use `THREAD_OFF`).

Reply:
`thread_cancel` checks for a NULL pointer. `THREAD_OFF` is deprecated.

Reply:
Indeed. I wasn't aware of that API change.